src/bci_build/package/appcontainers.py

```diff
@@ -271,6 +271,80 @@
 ]
 
 
+def _get_postfix_kwargs(os_version: OsVersion):
+    kwargs = {
+        "os_version": os_version,
+        "is_latest": os_version in CAN_BE_LATEST_OS_VERSION,
+        "version": "%%postfix_version%%",
+        "version_in_uid": False,
+        "replacements_via_service": [
+            Replacement(
+                regex_in_build_description="%%postfix_version%%",
+                package_name="postfix",
+                parse_version="minor",
+            )
+        ],
+        "package_list": [
+            "bash",
+            "postfix",
+            "postfix-ldap",
+            "postfix-lmdb",
+            "mandoc",
+            "netcfg",
+            "cyrus-sasl",
+            "cyrus-sasl-plain",
+            "ca-certificates",
+            "ca-certificates-mozilla",
+            "spamassassin-spamc",
+            "spamass-milter",
+            "timezone",
+            "-busybox-man",
+            "-busybox-sendmail",
+            "-busybox-iproute2",
+        ],
+        "entrypoint": ["/entrypoint/entrypoint.sh"],
+        "cmd": ["postfix", "start"],
+        "build_recipe_type": BuildType.DOCKER,
+        "extra_files": _POSTFIX_FILES,
+        "support_level": SupportLevel.L3,  # TODO: confirm support level
+        "exposes_tcp": [25, 465, 587],
+        "volumes": ["/var/spool/postfix", "/var/spool/vmail", "/etc/pki"],
+    }
+
+    return kwargs
+
+
+_POSTFIX_FILES = {}
+for filename in (
+    "entrypoint.sh",
+    "smtpd_sender_login_maps",
+    "virtual_alias_domains",
+    "virtual_alias_maps",
+    "virtual_gid_maps",
+    "virtual_mailbox_maps",
+    "virtual_uid_maps",
+):
+    if filename == "entrypoint.sh":
+        _POSTFIX_FILES[filename] = (
+            Path(__file__).parent / "postfix" / "entrypoint" / filename
+        ).read_bytes()
+    else:
+        _POSTFIX_FILES[filename] = (
+            Path(__file__).parent / "postfix" / "entrypoint" / "ldap" / filename
+        ).read_bytes()
+
+POSTFIX_CONTAINERS = [
+    ApplicationStackContainer(
+        name="postfix",
+        package_name="postfix-image",
+        pretty_name="Postfix",
+        custom_description="Postfix container as fast and secure mail server, {based_on_container}.",
+        **_get_postfix_kwargs(os_version),
```

Comment:
Any particular reason why you opted into putting the kwargs creation into a function and not just explicitly listing them here? Using the indirection via a function usually confuses type checkers & linters and I don't see an immediate benefit here

Reply:
No specific reason. I was using the nginx containers (for reference) & ended up copy/pasting.

Listed them explicitly now.